Projects/UOContent/Items/Misc/OilFlask.cs

```diff
@@ -0,0 +1,98 @@
+using ModernUO.Serialization;
+using Server.Targeting;
+
+namespace Server.Items;
+
+[SerializationGenerator(0)]
+public partial class OilFlask : Item
+{
+    [Constructible]
+    public OilFlask(int amount = 1) : base(0x1C18)
+    {
+        Stackable = true;
+        Amount = amount;
+    }
+
+    public override int LabelNumber => 1027199; // oil flask
+    public virtual int FilledMessageNumber => 1150864; // You fill the lamp with oil.
+    public virtual double FillMultiplier => 1;
+
+    public override void OnDoubleClick(Mobile from)
+    {
+        from.SendLocalizedMessage(501947); // Select lantern to refuel.
+        from.BeginTarget(2, true, TargetFlags.None, OnTargetPicked);
+    }
+
+    private void OnTargetPicked(Mobile from, object targeted)
+    {
+        switch (targeted)
```

Comment:
While this works, let's use `if (targeted is Lantern lantern) {} else {}` instead.